ee/server/service/software_installers.go

```diff
@@ -1501,12 +1501,13 @@ func (svc *Service) BatchSetSoftwareInstallers(
 				fmt.Sprintf("Couldn't edit software. URL (%q) is invalid", payload.URL),
 			)
 		}
-		validatedLabels, err := ValidateSoftwareLabels(ctx, svc, payload.LabelsIncludeAny, payload.LabelsExcludeAny)
-		if err != nil {
-			return "", err
+		if !dryRun {
+			validatedLabels, err := ValidateSoftwareLabels(ctx, svc, payload.LabelsIncludeAny, payload.LabelsExcludeAny)
+			if err != nil {
+				return "", err
+			}
+			payload.ValidatedLabels = validatedLabels
 		}
-		payload.ValidatedLabels = validatedLabels
```

Comment:
`payload.ValidatedLabels` is a pointer, and there's `nil` checks for it, so it's ok if this isn't populated in dry run mode.